service/history/queues/executable.go

```diff
@@ -54,3 +66,250 @@ type (
+	}
+)
+
+func NewExecutable(
+	task tasks.Task,
+	filter TaskFilter,
+	executor Executor,
+	scheduler Scheduler,
+	rescheduler Rescheduler,
+	timeSource clock.TimeSource,
+	logger log.Logger,
+	scope metrics.Scope,
+	criticalRetryCount dynamicconfig.IntPropertyFn,
+	queueType QueueType,
+) Executable {
+	return &executableImpl{
+		Task:               task,
+		state:              ctasks.TaskStatePending,
+		attempt:            1,
+		executor:           executor,
+		scheduler:          scheduler,
+		rescheduler:        rescheduler,
+		timeSource:         timeSource,
+		loadTime:           timeSource.Now(),
+		logger:             tasks.InitializeLogger(task, logger),
+		scope:              scope,
+		queueType:          queueType,
+		criticalRetryCount: criticalRetryCount,
+		filter:             filter,
+	}
+}
+
+func (e *executableImpl) Execute() error {
+	// this filter should also contain the logic for overriding
+	// results from task allocator (force executing some standby task types)
+	e.shouldProcess = e.filter(e.Task)
+	if !e.shouldProcess {
+		return nil
+	}
+
+	ctx := metrics.AddMetricsContext(context.Background())
+	startTime := e.timeSource.Now()
+	err := e.executor.Execute(ctx, e)
+	var userLatency time.Duration
+	if duration, ok := metrics.ContextCounterGet(ctx, metrics.HistoryWorkflowExecutionCacheLatency); ok {
+		userLatency = time.Duration(duration)
+	}
+	e.userLatency += userLatency
+
+	e.scope.IncCounter(metrics.TaskRequests)
+	e.scope.RecordTimer(metrics.TaskProcessingLatency, time.Since(startTime))
+	e.scope.RecordTimer(metrics.TaskNoUserProcessingLatency, time.Since(startTime)-userLatency)
+	return err
+}
+
+func (e *executableImpl) HandleErr(err error) (retErr error) {
+	defer func() {
+		if retErr != nil {
+			e.Lock()
+			defer e.Unlock()
+
+			e.attempt++
+			if e.attempt > e.criticalRetryCount() {
+				e.scope.RecordDistribution(metrics.TaskAttemptTimer, e.attempt)
+				e.logger.Error("Critical error processing task, retrying.", tag.Error(err), tag.OperationCritical)
+			}
+		}
+	}()
+
+	if err == nil {
+		return nil
+	}
+
+	if _, ok := err.(*serviceerror.NotFound); ok {
+		return nil
+	}
+
+	if err == consts.ErrTaskRetry {
+		e.scope.IncCounter(metrics.TaskStandbyRetryCounter)
+		return err
+	}
+
+	if err == consts.ErrTaskDiscarded {
+		e.scope.IncCounter(metrics.TaskDiscarded)
+		return nil
+	}
+
+	// this is a transient error
+	// TODO remove this error check special case
+	//  since the new task life cycle will not give up until task processed / verified
+	if _, ok := err.(*serviceerror.NamespaceNotActive); ok {
+		if e.timeSource.Now().Sub(e.loadTime) > 2*namespace.CacheRefreshInterval {
+			e.scope.IncCounter(metrics.TaskNotActiveCounter)
+			return nil
+		}
+
+		return err
+	}
+
+	e.scope.IncCounter(metrics.TaskFailures)
+
+	if _, ok := err.(*persistence.CurrentWorkflowConditionFailedError); ok {
+		e.logger.Error("More than 2 workflow are running.", tag.Error(err), tag.LifeCycleProcessingFailed)
+		return nil
+	}
```

Comment:
you can remove this check

Reply:
Some additional context here, the error will only happen if there's bug in the system and in that case, we should not ignore the issue.